src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class Lottery {
+    private int randomNumber;
+    private String randomColor;
+
+    public Ball getRandomBall() {
+        return new Ball(getRandomNumber(),getRandomColor());
+    }
+
+    private void setRandomColor() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        this.randomColor = colorSupplier.getRandomColor();
+    }
+
+    private void setRandomNumber() {
+        NumberSupplier numberSupplier = new NumberSupplier();
+        this.randomNumber = numberSupplier.getRandomNumber();
+    }
+
+    private int getRandomNumber() {
+        setRandomNumber();
+        return randomNumber;
+    }
+
+    private String getRandomColor() {
+        setRandomColor();
+        return randomColor;
```

Comment:
1. Services we rely on are better as class variable in order not to create new object each time method is called.
2. Same as in comment above, class variables and private mothods are not required:
```suggestion
    private final ColorSupplier = new ColorSupplier();
    private final NumberSupplier = new NumberSupplier();
    
    public Ball getRandomBall() {
        return new Ball(numberSupplier.getRandomNumber(), colorSupplier.getRandomColor());
 
```